system/lib/wasmfs/js_api.cpp

```diff
@@ -246,4 +256,11 @@ void _wasmfs_readdir_finish(struct wasmfs_readdir_state* state) {
   free(state);
 }
 
+char* _wasmfs_get_cwd(void) {
+  // TODO: PATH_MAX is 4K atm, so it might be good to reduce this somehow.
```

Comment:
One possible solution here would be to call getcwd + realloc in a loop as long as getcwd returns ENAMETOOLONG.. eventually the buffer will then reach the correct size.

Not too important though I think.

Reply:
Either is ok I guess. Good to have that option if we find out we run into issues with the leaking here.